homeassistant/components/plugwise/sensor.py

```diff
@@ -0,0 +1,345 @@
+
+                    if "plug" in device_properties["types"]:
+                        model = "Metered Switch"
+
+                    entities.append(
+                        PwPowerSensor(
+                            api,
+                            coordinator,
+                            device_properties["name"],
+                            dev_id,
+                            sensor,
+                            sensor_type,
+                            model,
+                        )
+                    )
+                else:
+                    entities.append(
+                        PwThermostatSensor(
+                            api,
+                            coordinator,
+                            device_properties["name"],
+                            dev_id,
+                            sensor,
+                            sensor_type,
+                        )
+                    )
+
+        if single_thermostat is False:
+            for state in INDICATE_ACTIVE_LOCAL_DEVICE:
+                if state in data:
+                    entities.append(
+                        PwAuxDeviceSensor(
+                            api,
+                            coordinator,
+                            device_properties["name"],
+                            dev_id,
+                            DEVICE_STATE,
+                        )
+                    )
+                    break
+
+    async_add_entities(entities, True)
+
+
+class SmileSensor(SmileGateway):
+    """Represent Smile Sensors."""
+
+    def __init__(self, api, coordinator, name, dev_id, sensor):
+        """Initialise the sensor."""
+        super().__init__(api, coordinator, name, dev_id)
+
+        self._sensor = sensor
+
+        self._dev_class = None
+        self._state = None
+        self._unit_of_measurement = None
+
+        if dev_id == self._api.heater_id:
+            self._entity_name = "Auxiliary"
+
+        sensorname = sensor.replace("_", " ").title()
+        self._name = f"{self._entity_name} {sensorname}"
+
+        if dev_id == self._api.gateway_id:
+            self._entity_name = f"Smile {self._entity_name}"
+
+        self._unique_id = f"{dev_id}-{sensor}"
+
+    @property
+    def device_class(self):
+        """Device class of this entity."""
+        return self._dev_class
+
+    @property
+    def state(self):
+        """Device class of this entity."""
+        return self._state
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement of this entity, if any."""
+        return self._unit_of_measurement
+
+
+class PwThermostatSensor(SmileSensor, Entity):
+    """Thermostat and climate sensor entities."""
+
+    def __init__(self, api, coordinator, name, dev_id, sensor, sensor_type):
+        """Set up the Plugwise API."""
+        super().__init__(api, coordinator, name, dev_id, sensor)
+
+        self._api = api
+        self._sensor_type = sensor_type
```

Comment:
`sensor_type` only seems to be used locally in this method. If so we don't need to store it as an instance attribute.